File: Basic/Count numbers containing 4.cpp
 
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
class Solution
{
  public:
    int countNumberswith4(int N)
    {
        // code here
        
        int ans = 0;
        
        for (int i=0; i<=N; i++)
        {
            string s = to_string(i);
            
            for (int j=0; j<s.length(); j++)
            {
                if (s.at(j) == '4')
                {
                    ans++;
                    break;
                }
            
            }
            
        }
        
        
        return ans;
        
    }
};

// { Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        
        cin>>N;

        Solution ob;
        cout << ob.countNumberswith4(N) << endl;
    }
    return 0;
}  // } Driver Code Ends

File: Easy/Count numbers containing 4.cs
//{ Driver Code Starts
using System;

namespace DriverCode {
class GFG {
    static void Main(string[] args) {
        int testcases;
        testcases = Convert.ToInt32(Console.ReadLine());
        while (testcases-- > 0) {

            int n = Convert.ToInt32(Console.ReadLine().Trim());

            Solution obj = new Solution();
            int res = obj.countNumberswith4(n);
            Console.WriteLine(res);
        }
    }
}
}

// } Driver Code Ends

class Solution
{
    public int countNumberswith4(int n) 
    {
        // code here
        int ans = 0;
        
        for (int i = 4; i <= n; i++)
        {
            int j = i;
            
            while (j > 0)
            {
                if (j % 10 == 4)
                {
                    ans++;
                    break;
                }
                
                j /= 10;
            }
            
        }
        
        return ans;
        
    }
}
